sycl/test/devicelib/assert.cpp

```diff
@@ -0,0 +1,200 @@
+// REQUIRES: cpu,linux
+// RUN: %clangxx -fsycl -c %s -o %t.o
+// RUN: %clangxx -fsycl %t.o %llvm_build_libs_dir/libsycl-glibc.o -o %t.out
+// (see the other RUN lines below; it is a bit complicated)
+//
+// assert() call in device code guarantees nothing: on some devices it behaves
+// in the usual way and terminates the program. On other devices it can print an
+// error message and *continue* execution. Less capable devices can even ignore
+// an assert!
+//
+// This makes testing an assert() a bit difficult task, and we have to rely on
+// the implementation details to make sure that both "native" and "fallback"
+// implementations work as expected.
+//
+// This test works only on Intel OpenCL CPU implementation, which is known to
+// behave as follows:
+//
+//   Fallback mode (aka the best we can do by following the OpenCL spec):
+//     1. Assertion condition is printed to *stdout* by the OpenCL printf().
+//     2. Process (both host and device) is terminated by a SIGSEGV.
+//
+//   Native mode (same behavior as libc assert on CPU):
+//     1. Assertion condition is printed to *stderr*.
+//     2. Process (both host and device) is terminated by a SIGABRT.
```

Comment:
So the fallback mode is not the native mode?

Reply:
Or are you thinking to some OpenCL native kernels that could be used in SYCL interoperability mode or even by some OpenCL compiler for x86? https://www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/clEnqueueNativeKernel.html

Reply:
> So the fallback mode is not the native mode?

Basically, we have not 2, but 3 modes:
1. Host device, where everything works already (it is a standard C++ code after all)
2. Fallback mode (on OpenCL/PI device), which is supposed to work with any device
3. Native mode (on OpenCL/PI device) provides the best possible implementation for a particular device.

Reply:
OK, this seems different from what OpenCL defines as "native" then.